xl/player/player.py

```diff
@@ -187,15 +195,48 @@ def play(self, track, start_at=None, paused=False):
         if track is None:
             self.stop()
         else:
+            self.stop()
             if self.is_stopped():
                 event.log_event('playback_player_start', self, track)
 
-            play_args = self._get_play_params(track, start_at, paused, False)
-            self._engine.play(*play_args)
-            if play_args[2]:
+            self._current_play_args = self._get_play_params(
+                track, start_at, paused, False
+            )
+
+            self._delayed_until = 0
+            self._cancel_delayed_stop()
+            self._track = track
+            start_at = self._current_play_args[1]
+
+            if start_at < 0:
+                logger.debug('entering delayed start')
+                self._delayed_until = -1 * start_at + time.time()
+                self.engine_notify_track_start(track)
+                self._delay_id = GLib.timeout_add_seconds(1, self._play_engine)
+            else:
+                self._play_engine()
+
+            if self._current_play_args[2]:
                 event.log_event('playback_player_pause', self, track)
```

Comment:
This feels like it should happen in _play_engine instead? I guess I don't really remember what this is supposed to do.

Reply:
This is used in case that there is a global auto advance delay. Then the player is paused until the delay times out.
I'll try to move.